From bb59c7225e2374b9cf732c98a6f71d3e9436c4ae Mon Sep 17 00:00:00 2001
From: Shammah Chancellor <shammah.chancellor@notvalid.com>
Date: Wed, 14 Oct 2026 07:37:02 -0700
Subject: [PATCH] tests: give large amount compression literals an explicit
 int64_t type

On Apple arm64, unsuffixed decimal literals above INT_MAX have type long, while int64_t is long long. Multiplication by Amount then ambiguously matches its int and int64_t overloads. Explicit fixed-width literals preserve the tested amounts and allow the Dogecoin compression tests to compile without changing arithmetic or consensus.
---
 src/test/compress_tests.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/test/compress_tests.cpp b/src/test/compress_tests.cpp
index 65e1cbba0b664e29e91816c82d65986f2b2dce47..73e6954c5d8dada93c81b0ade8964ab0de1912d9 100644
--- a/src/test/compress_tests.cpp
+++ b/src/test/compress_tests.cpp
@@ -47,9 +47,9 @@ BOOST_AUTO_TEST_CASE(compress_amounts) {
     BOOST_CHECK(TestPair(COIN, 0x9));
     BOOST_CHECK(TestPair(50 * COIN, 0x32));
     BOOST_CHECK(TestPair(21000000 * COIN, 0x1406f40));
-    BOOST_CHECK(TestPair(10000000000 * COIN, 0x2540be400));
+    BOOST_CHECK(TestPair(int64_t{10000000000} * COIN, 0x2540be400));
     BOOST_CHECK(TestPair(MAX_MONEY, 0x2540be400));
-    BOOST_CHECK(TestPair(10000000001 * COIN, 0x14F46B0409));
+    BOOST_CHECK(TestPair(int64_t{10000000001} * COIN, 0x14F46B0409));
     BOOST_CHECK(TestPair(MAX_MONEY + COIN, 0x14F46B0409));
 
     for (int64_t i = 1; i <= NUM_MULTIPLES_UNIT; i++) {
